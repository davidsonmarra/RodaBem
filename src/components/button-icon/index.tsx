import React from 'react';
import {TouchableOpacity, TouchableOpacityProps} from 'react-native';
import styles from './styles';

interface ButtonIconProps extends TouchableOpacityProps {}

const ButtonIcon = ({children, onPress}: ButtonIconProps) => {
  return (
    <TouchableOpacity onPress={onPress} style={styles.container}>
      {children}
    </TouchableOpacity>
  );
};

export default ButtonIcon;
