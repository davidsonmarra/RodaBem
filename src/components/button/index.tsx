import React from 'react';
import {TouchableOpacity, TouchableOpacityProps} from 'react-native';
import styles from './styles';
import Text, {TextType} from '../text';

export enum ButtonType {
  primary = 'primary',
  secondary = 'secondary',
}

export interface Props extends TouchableOpacityProps {
  text: string;
  type?: ButtonType;
}

const buttonProps = {
  [ButtonType.primary]: {
    buttonStyle: styles.primary,
    textType: TextType.buttonPrimary,
  },
  [ButtonType.secondary]: {
    buttonStyle: styles.secondary,
    textType: TextType.buttonSecondary,
  },
};

const Button = ({text, type = ButtonType.primary, ...rest}: Props) => {
  return (
    <TouchableOpacity
      style={[styles.container, buttonProps[type].buttonStyle, rest.style]}
      {...rest}>
      <Text type={buttonProps[type].textType}>{text}</Text>
    </TouchableOpacity>
  );
};

export default Button;
