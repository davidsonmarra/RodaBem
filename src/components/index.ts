export {default as Button, ButtonType} from './button';
export {default as ButtonIcon} from './button-icon';
export {default as Header} from './header';
export {default as HorizontalSpacer} from './horizontal-spacer';
export {default as Input} from './input';
export {default as StorybookView} from './storybook-view';
export {default as Text, TextType} from './text';
export {default as VerticalSpacer} from './vertical-spacer';
