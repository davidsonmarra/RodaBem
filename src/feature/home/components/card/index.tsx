import React from 'react';
import {TouchableOpacity, TouchableOpacityProps, View} from 'react-native';
import {Text, TextType, VerticalSpacer} from '@components';
import styles from './styles';

interface Props extends TouchableOpacityProps {
  data: {
    name: string;
    brand: string;
    kmDriven?: string;
    kmPerLiter: string;
  };
}

const Card = ({data, onPress}: Props) => {
  return (
    <TouchableOpacity onPress={onPress} style={styles.container}>
      <View style={styles.leftContent}>
        <Text type={TextType.title}>{data.name}</Text>
        <Text type={TextType.text}>{data.brand}</Text>
        <VerticalSpacer height={12} />
        {data.kmDriven && (
          <Text type={TextType.text}>{`${data.kmDriven} Km rodados`}</Text>
        )}
      </View>
      <View style={styles.rightContent}>
        <Text type={TextType.title}>{`${data.kmDriven} Km/L`}</Text>
      </View>
    </TouchableOpacity>
  );
};

export default Card;
