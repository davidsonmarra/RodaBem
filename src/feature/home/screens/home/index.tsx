import React from 'react';
import {NativeStackScreenProps} from '@react-navigation/native-stack';
import HomeContainer from './ui';
import {RootStackParamList} from 'src/feature/home/navigation';

type NavigationProps = NativeStackScreenProps<RootStackParamList, 'Home'>;

interface Props extends NavigationProps {}

const HomeScreen = ({navigation}: Props) => {
  const handleCardPress = () => {
    navigation.navigate('Test');
  };

  const handleAddCarPress = () => {
    navigation.navigate('Test');
  };

  return (
    <HomeContainer
      onCardPress={handleCardPress}
      onAddCarPress={handleAddCarPress}
    />
  );
};

export default HomeScreen;
