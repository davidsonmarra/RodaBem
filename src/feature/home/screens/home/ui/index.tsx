import React from 'react';
import {FlatList, SafeAreaView, View} from 'react-native';
import {Header, VerticalSpacer} from '@components';
import {Card} from 'src/feature/home/components';
import styles from './styles';
import ButtonIcon from 'src/components/button-icon';
import {AddIcon} from '@assets';

const data = [
  {
    name: 'Name',
    brand: 'Brand',
    kmDriven: '10',
    kmPerLiter: '13',
  },
  {
    name: 'Name',
    brand: 'Brand',
    kmDriven: '10',
    kmPerLiter: '13',
  },
  {
    name: 'Name',
    brand: 'Brand',
    kmDriven: '10',
    kmPerLiter: '13',
  },
  {
    name: 'Name',
    brand: 'Brand',
    kmDriven: '10',
    kmPerLiter: '13',
  },
  {
    name: 'Name',
    brand: 'Brand',
    kmDriven: '10',
    kmPerLiter: '13',
  },
  {
    name: 'Name',
    brand: 'Brand',
    kmDriven: '10',
    kmPerLiter: '13',
  },
  {
    name: 'Name',
    brand: 'Brand',
    kmDriven: '10',
    kmPerLiter: '13',
  },
  {
    name: 'Name',
    brand: 'Brand',
    kmDriven: '10',
    kmPerLiter: '13',
  },
  {
    name: 'Name',
    brand: 'Brand',
    kmDriven: '10',
    kmPerLiter: '13',
  },
];

interface Props {
  onCardPress: () => void;
  onAddCarPress: () => void;
}

const HomeContainer = ({onCardPress, onAddCarPress}: Props) => {
  return (
    <SafeAreaView style={styles.container}>
      <Header text="Carros" showLeftIcon={false} />
      <View style={styles.content}>
        <FlatList
          data={data}
          renderItem={({item}) => <Card onPress={onCardPress} data={item} />}
          keyExtractor={(item, index) => index.toString()}
          showsVerticalScrollIndicator={false}
          ItemSeparatorComponent={() => <VerticalSpacer height={12} />}
          contentContainerStyle={{paddingBottom: 64 + 24}}
        />
      </View>
      <View style={styles.buttonContainer}>
        <ButtonIcon onPress={onAddCarPress}>
          <AddIcon />
        </ButtonIcon>
      </View>
    </SafeAreaView>
  );
};

export default HomeContainer;
