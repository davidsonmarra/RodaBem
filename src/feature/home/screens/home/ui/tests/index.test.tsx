import React from 'react';
import {create} from 'react-test-renderer';
import {ButtonIcon} from '@components';
import HomeContainer from '..';
import {Card} from 'src/feature/home/components';

const mockOnAddCarPress = jest.fn();
const mockOnCardPress = jest.fn();

jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({
    navigate: jest.fn(),
    goBack: jest.fn(),
  }),
}));

const containerInstance = ({
  onAddCarPress = mockOnAddCarPress,
  onCardPress = mockOnCardPress,
}: any) =>
  create(
    <HomeContainer onAddCarPress={onAddCarPress} onCardPress={onCardPress} />,
  );

describe('SignInContainer', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const tree = containerInstance({}).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call onAddCarPress', () => {
    const instance = containerInstance({}).root;
    const buttonsInstance = instance.findAllByType(ButtonIcon);
    buttonsInstance[0].props.onPress();
    expect(mockOnAddCarPress).toHaveBeenCalledTimes(1);
  });

  it('should call onCardPress', () => {
    const instance = containerInstance({}).root;
    const buttonsInstance = instance.findAllByType(Card);
    buttonsInstance[0].props.onPress();
    expect(mockOnCardPress).toHaveBeenCalledTimes(1);
  });
});
