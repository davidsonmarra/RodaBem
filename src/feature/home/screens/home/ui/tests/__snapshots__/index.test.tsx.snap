// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInContainer should render correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#F1F1F1",
      "flex": 1,
      "paddingHorizontal": 24,
      "paddingVertical": 16,
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "paddingHorizontal": 12,
        "paddingVertical": 16,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "width": 24,
        }
      }
    />
    <View
      style={
        {
          "width": 12,
        }
      }
    />
    <Text
      style={
        [
          {
            "color": "#121212",
            "fontSize": 18,
            "fontWeight": "bold",
          },
          {
            "flex": 1,
            "textAlign": "center",
          },
        ]
      }
    >
      Carros
    </Text>
    <View
      style={
        {
          "width": 12,
        }
      }
    />
    <View
      style={
        {
          "width": 24,
        }
      }
    />
  </View>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      contentContainerStyle={
        {
          "paddingBottom": 88,
        }
      }
      data={
        [
          {
            "brand": "Brand",
            "kmDriven": "10",
            "kmPerLiter": "13",
            "name": "Name",
          },
          {
            "brand": "Brand",
            "kmDriven": "10",
            "kmPerLiter": "13",
            "name": "Name",
          },
          {
            "brand": "Brand",
            "kmDriven": "10",
            "kmPerLiter": "13",
            "name": "Name",
          },
          {
            "brand": "Brand",
            "kmDriven": "10",
            "kmPerLiter": "13",
            "name": "Name",
          },
          {
            "brand": "Brand",
            "kmDriven": "10",
            "kmPerLiter": "13",
            "name": "Name",
          },
          {
            "brand": "Brand",
            "kmDriven": "10",
            "kmPerLiter": "13",
            "name": "Name",
          },
          {
            "brand": "Brand",
            "kmDriven": "10",
            "kmPerLiter": "13",
            "name": "Name",
          },
          {
            "brand": "Brand",
            "kmDriven": "10",
            "kmPerLiter": "13",
            "name": "Name",
          },
          {
            "brand": "Brand",
            "kmDriven": "10",
            "kmPerLiter": "13",
            "name": "Name",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={0.0001}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#E5E5E5",
                "borderRadius": 8,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "opacity": 1,
                "paddingHorizontal": 12,
                "paddingVertical": 16,
              }
            }
          >
            <View
              style={
                {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#121212",
                      "fontSize": 24,
                      "fontWeight": "bold",
                    },
                    undefined,
                  ]
                }
              >
                Name
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#222222",
                      "fontSize": 16,
                      "fontWeight": "normal",
                    },
                    undefined,
                  ]
                }
              >
                Brand
              </Text>
              <View
                style={
                  {
                    "height": 12,
                  }
                }
              />
              <Text
                style={
                  [
                    {
                      "color": "#222222",
                      "fontSize": 16,
                      "fontWeight": "normal",
                    },
                    undefined,
                  ]
                }
              >
                10 Km rodados
              </Text>
            </View>
            <View
              style={
                {
                  "marginLeft": 8,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#121212",
                      "fontSize": 24,
                      "fontWeight": "bold",
                    },
                    undefined,
                  ]
                }
              >
                10 Km/L
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "height": 12,
              }
            }
          />
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#E5E5E5",
                "borderRadius": 8,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "opacity": 1,
                "paddingHorizontal": 12,
                "paddingVertical": 16,
              }
            }
          >
            <View
              style={
                {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#121212",
                      "fontSize": 24,
                      "fontWeight": "bold",
                    },
                    undefined,
                  ]
                }
              >
                Name
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#222222",
                      "fontSize": 16,
                      "fontWeight": "normal",
                    },
                    undefined,
                  ]
                }
              >
                Brand
              </Text>
              <View
                style={
                  {
                    "height": 12,
                  }
                }
              />
              <Text
                style={
                  [
                    {
                      "color": "#222222",
                      "fontSize": 16,
                      "fontWeight": "normal",
                    },
                    undefined,
                  ]
                }
              >
                10 Km rodados
              </Text>
            </View>
            <View
              style={
                {
                  "marginLeft": 8,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#121212",
                      "fontSize": 24,
                      "fontWeight": "bold",
                    },
                    undefined,
                  ]
                }
              >
                10 Km/L
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "height": 12,
              }
            }
          />
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#E5E5E5",
                "borderRadius": 8,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "opacity": 1,
                "paddingHorizontal": 12,
                "paddingVertical": 16,
              }
            }
          >
            <View
              style={
                {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#121212",
                      "fontSize": 24,
                      "fontWeight": "bold",
                    },
                    undefined,
                  ]
                }
              >
                Name
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#222222",
                      "fontSize": 16,
                      "fontWeight": "normal",
                    },
                    undefined,
                  ]
                }
              >
                Brand
              </Text>
              <View
                style={
                  {
                    "height": 12,
                  }
                }
              />
              <Text
                style={
                  [
                    {
                      "color": "#222222",
                      "fontSize": 16,
                      "fontWeight": "normal",
                    },
                    undefined,
                  ]
                }
              >
                10 Km rodados
              </Text>
            </View>
            <View
              style={
                {
                  "marginLeft": 8,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#121212",
                      "fontSize": 24,
                      "fontWeight": "bold",
                    },
                    undefined,
                  ]
                }
              >
                10 Km/L
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "height": 12,
              }
            }
          />
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#E5E5E5",
                "borderRadius": 8,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "opacity": 1,
                "paddingHorizontal": 12,
                "paddingVertical": 16,
              }
            }
          >
            <View
              style={
                {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#121212",
                      "fontSize": 24,
                      "fontWeight": "bold",
                    },
                    undefined,
                  ]
                }
              >
                Name
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#222222",
                      "fontSize": 16,
                      "fontWeight": "normal",
                    },
                    undefined,
                  ]
                }
              >
                Brand
              </Text>
              <View
                style={
                  {
                    "height": 12,
                  }
                }
              />
              <Text
                style={
                  [
                    {
                      "color": "#222222",
                      "fontSize": 16,
                      "fontWeight": "normal",
                    },
                    undefined,
                  ]
                }
              >
                10 Km rodados
              </Text>
            </View>
            <View
              style={
                {
                  "marginLeft": 8,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#121212",
                      "fontSize": 24,
                      "fontWeight": "bold",
                    },
                    undefined,
                  ]
                }
              >
                10 Km/L
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "height": 12,
              }
            }
          />
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#E5E5E5",
                "borderRadius": 8,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "opacity": 1,
                "paddingHorizontal": 12,
                "paddingVertical": 16,
              }
            }
          >
            <View
              style={
                {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#121212",
                      "fontSize": 24,
                      "fontWeight": "bold",
                    },
                    undefined,
                  ]
                }
              >
                Name
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#222222",
                      "fontSize": 16,
                      "fontWeight": "normal",
                    },
                    undefined,
                  ]
                }
              >
                Brand
              </Text>
              <View
                style={
                  {
                    "height": 12,
                  }
                }
              />
              <Text
                style={
                  [
                    {
                      "color": "#222222",
                      "fontSize": 16,
                      "fontWeight": "normal",
                    },
                    undefined,
                  ]
                }
              >
                10 Km rodados
              </Text>
            </View>
            <View
              style={
                {
                  "marginLeft": 8,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#121212",
                      "fontSize": 24,
                      "fontWeight": "bold",
                    },
                    undefined,
                  ]
                }
              >
                10 Km/L
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "height": 12,
              }
            }
          />
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#E5E5E5",
                "borderRadius": 8,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "opacity": 1,
                "paddingHorizontal": 12,
                "paddingVertical": 16,
              }
            }
          >
            <View
              style={
                {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#121212",
                      "fontSize": 24,
                      "fontWeight": "bold",
                    },
                    undefined,
                  ]
                }
              >
                Name
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#222222",
                      "fontSize": 16,
                      "fontWeight": "normal",
                    },
                    undefined,
                  ]
                }
              >
                Brand
              </Text>
              <View
                style={
                  {
                    "height": 12,
                  }
                }
              />
              <Text
                style={
                  [
                    {
                      "color": "#222222",
                      "fontSize": 16,
                      "fontWeight": "normal",
                    },
                    undefined,
                  ]
                }
              >
                10 Km rodados
              </Text>
            </View>
            <View
              style={
                {
                  "marginLeft": 8,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#121212",
                      "fontSize": 24,
                      "fontWeight": "bold",
                    },
                    undefined,
                  ]
                }
              >
                10 Km/L
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "height": 12,
              }
            }
          />
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#E5E5E5",
                "borderRadius": 8,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "opacity": 1,
                "paddingHorizontal": 12,
                "paddingVertical": 16,
              }
            }
          >
            <View
              style={
                {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#121212",
                      "fontSize": 24,
                      "fontWeight": "bold",
                    },
                    undefined,
                  ]
                }
              >
                Name
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#222222",
                      "fontSize": 16,
                      "fontWeight": "normal",
                    },
                    undefined,
                  ]
                }
              >
                Brand
              </Text>
              <View
                style={
                  {
                    "height": 12,
                  }
                }
              />
              <Text
                style={
                  [
                    {
                      "color": "#222222",
                      "fontSize": 16,
                      "fontWeight": "normal",
                    },
                    undefined,
                  ]
                }
              >
                10 Km rodados
              </Text>
            </View>
            <View
              style={
                {
                  "marginLeft": 8,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#121212",
                      "fontSize": 24,
                      "fontWeight": "bold",
                    },
                    undefined,
                  ]
                }
              >
                10 Km/L
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "height": 12,
              }
            }
          />
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#E5E5E5",
                "borderRadius": 8,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "opacity": 1,
                "paddingHorizontal": 12,
                "paddingVertical": 16,
              }
            }
          >
            <View
              style={
                {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#121212",
                      "fontSize": 24,
                      "fontWeight": "bold",
                    },
                    undefined,
                  ]
                }
              >
                Name
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#222222",
                      "fontSize": 16,
                      "fontWeight": "normal",
                    },
                    undefined,
                  ]
                }
              >
                Brand
              </Text>
              <View
                style={
                  {
                    "height": 12,
                  }
                }
              />
              <Text
                style={
                  [
                    {
                      "color": "#222222",
                      "fontSize": 16,
                      "fontWeight": "normal",
                    },
                    undefined,
                  ]
                }
              >
                10 Km rodados
              </Text>
            </View>
            <View
              style={
                {
                  "marginLeft": 8,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#121212",
                      "fontSize": 24,
                      "fontWeight": "bold",
                    },
                    undefined,
                  ]
                }
              >
                10 Km/L
              </Text>
            </View>
          </View>
          <View
            style={
              {
                "height": 12,
              }
            }
          />
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#E5E5E5",
                "borderRadius": 8,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "opacity": 1,
                "paddingHorizontal": 12,
                "paddingVertical": 16,
              }
            }
          >
            <View
              style={
                {
                  "flex": 1,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#121212",
                      "fontSize": 24,
                      "fontWeight": "bold",
                    },
                    undefined,
                  ]
                }
              >
                Name
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#222222",
                      "fontSize": 16,
                      "fontWeight": "normal",
                    },
                    undefined,
                  ]
                }
              >
                Brand
              </Text>
              <View
                style={
                  {
                    "height": 12,
                  }
                }
              />
              <Text
                style={
                  [
                    {
                      "color": "#222222",
                      "fontSize": 16,
                      "fontWeight": "normal",
                    },
                    undefined,
                  ]
                }
              >
                10 Km rodados
              </Text>
            </View>
            <View
              style={
                {
                  "marginLeft": 8,
                }
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#121212",
                      "fontSize": 24,
                      "fontWeight": "bold",
                    },
                    undefined,
                  ]
                }
              >
                10 Km/L
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "bottom": 16,
        "position": "absolute",
        "right": 24,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#E30B0B",
          "borderRadius": 43,
          "height": 64,
          "justifyContent": "center",
          "opacity": 1,
          "width": 64,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="48"
        bbWidth="48"
        fill="#FFFFFF"
        focusable={false}
        height="48"
        meetOrSlice={0}
        minX={0}
        minY={-960}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 48,
              "width": 48,
            },
          ]
        }
        vbHeight={960}
        vbWidth={960}
        width="48"
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M440-440H200v-80h240v-240h80v240h240v80H520v240h-80v-240z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</RCTSafeAreaView>
`;
